pkg/providers/amifamily/ami.go

```diff
@@ -160,8 +160,30 @@ func (p *Provider) getDefaultAMIFromSSM(ctx context.Context, nodeTemplate *v1alp
 		if err != nil {
 			return nil, err
 		}
-		amis = append(amis, AMI{Name: ssmOutput.Name, AmiID: amiID, Requirements: ssmOutput.Requirements})
+		amis = append(amis, AMI{AmiID: amiID, Requirements: ssmOutput.Requirements})
 	}
+	// Creating selector filter by making a string of amiIds into a comma delineated string
+	ids := lo.Reduce(amis, func(agg string, item AMI, _ int) string {
+		return agg + item.AmiID + ","
+	}, "")
+	selector := map[string]string{"aws-ids": ids}
+	// Collecting the AMI details of the default AMIs from EC2
+	amisDetails, err := p.fetchAMIsFromEC2(ctx, selector)
+	if err != nil {
+		return nil, err
+	}
+
+	// matching up the AMIs details that is received from EC2 with the default AMIs
+	// collecting the names of the default AMIs
+	for i := range amis {
+		ami, ok := lo.Find(amisDetails, func(x *ec2.Image) bool {
+			return *x.ImageId == amis[i].AmiID
+		})
+		if ok {
+			amis[i].Name = *ami.Name
+		}
```

Comment:
Consider in-lining.
```suggestion
		if ami, ok := lo.Find(amisDetails, func(x *ec2.Image) bool {
			return *x.ImageId == amis[i].AmiID
		}) ; ok {
			amis[i].Name = *ami.Name
		}
```

You can also use lo.Map() so that you don't have to use a for loop if you'd like too.